plugins/drivers/driver.go

```diff
@@ -274,6 +274,7 @@ type TaskConfig struct {
 	JobName          string
 	JobID            string
 	TaskGroupName    string
+	ParentJobId      string
```

Comment:
```suggestion
	ParentJobID      string
```

We also will need to add this field to the `TaskConfig` message in [`driver.proto`](https://github.com/hashicorp/nomad/blob/main/plugins/drivers/proto/driver.proto). The reason this works without it is because the Docker driver is internal to Nomad so we don't use the protobuf messages to talk to it. But if we leave the new field out of the protobuf it would mean that it would be unavailable to external drivers like podman or containerd.